File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.util.*;
import java.io.*;

public class Main {

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(
                new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
        int V = Integer.parseInt(st.nextToken());

        graph = new boolean[N + 1][N + 1];

        for (int i = 0; i < M; i++) {
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            graph[a][b] = true;
            graph[b][a] = true;
        }
        visit = new boolean[N + 1];
        sb = new StringBuilder();
        dfs2(V);
        System.out.println(sb);
        visit = new boolean[N + 1];
        sb = new StringBuilder();
        bfs(V);
        System.out.println(sb);
    }

    static boolean[] visit;
    static boolean[][] graph;
    static StringBuilder sb;

    static void dfs(int node) {
        visit[node] = true;
        sb.append(node);
        sb.append(' ');
        for (int i = 1; i < graph[node].length; i++) {
            if (graph[node][i] && !visit[i]) {
                dfs(i);
            }
        }
    }

    static void dfs2(int node) {
        Stack<Integer> stack = new Stack<>();
        stack.push(node);
        while (!stack.isEmpty()) {
            int nextNode = stack.pop();
            if (visit[nextNode]) {
                continue;
            } else {
                visit[nextNode] = true;
            }
            sb.append(nextNode);
            sb.append(' ');
            for (int i = graph[nextNode].length - 1; i >= 1; i--) {
                if (!graph[nextNode][i]) {
                    continue;
                }
                stack.push(i);
            }
        }
    }

    static void bfs(int node) {
        Queue<Integer> queue = new ArrayDeque<>();
        queue.add(node);
        while (!queue.isEmpty()) {
            int nextNode = queue.poll();
            if (visit[nextNode]) {
                continue;
            } else {
                visit[nextNode] = true;
            }
            sb.append(nextNode);
            sb.append(' ');
            for (int i = 1; i <= graph[nextNode].length - 1; i++) {
                if (!graph[nextNode][i]) {
                    continue;
                }
                queue.add(i);
            }
        }
    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque
import sys
input = lambda: sys.stdin.readline().rstrip()

N, M, V = map(int, input().split())

graph = [[False] * (N+1) for _ in range(N+1)]

for i in range(M):
    a, b = map(int, input().split())
    graph[a][b] = True
    graph[b][a] = True

# DFS
def dfs2(node, visit):
    stack = deque()
    stack.append(node)
    while stack:
        next_node = stack.pop()
        if visit[next_node]: continue
        visit[next_node] = True
        print(next_node, end=' ')
        for i in range(len(visit)-1, 0, -1):
            if not graph[next_node][i]: continue
            stack.append(i)

# BFS
def bfs(node, visit):
    queue = deque()
    queue.appendleft(node)
    while queue:
        next_node = queue.pop()
        if visit[next_node]: continue
        visit[next_node] = True
        print(next_node, end=' ')
        for i in range(1, len(visit)):
            if not graph[next_node][i] : continue
            queue.appendleft(i)

dfs2(V, [False] * (N+1))
print()
bfs(V, [False] * (N+1))